scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -619,163 +814,255 @@ class ProcessMonitor(ScalyrMonitor):
-      app.uptime:                       the number of milliseconds of uptime
-      app.threads:                      the number of threads being used by the process
-      app.nice:                         the nice value for the process
-      app.mem.bytes type=vmsize:        the number of bytes of virtual memory in use
-      app.mem.bytes type=resident:      the number of bytes of resident memory in use
-      app.mem.bytes type=peak_vmsize:   the maximum number of bytes used for virtual memory for process
-      app.mem.bytes type=peak_resident: the maximum number of bytes of resident memory ever used by process
-      app.disk.bytes type=read:         the number of bytes read from disk
-      app.disk.requests type=read:      the number of disk requests.
-      app.disk.bytes type=write:        the number of bytes written to disk
-      app.disk.requests type=write:     the number of disk requests.
-      app.io.fds type=open:             the number of file descriptors held open by the process
 
     In additional to the fields listed above, each metric will also have a field 'app' set to the monitor id to specify
     which process the metric belongs to.
 
     You can run multiple instances of this monitor per agent to monitor different processes.
     """
+
     def _initialize(self):
         """Performs monitor-specific initialization."""
-        # The id of the process being monitored, if one has been matched.
-        self.__pid = None
-        # The list of BaseReaders instantiated to gather metrics for the process.
-        self.__gathers = []
+        # The ids of the processes being monitored, if one has been matched.
+        self.__pids = None
 
         self.__id = self._config.get('id', required_field=True, convert_to=str)
         self.__commandline_matcher = self._config.get('commandline', default=None, convert_to=str)
-        self.__target_pid = self._config.get('pid', default=None, convert_to=str)
-
-        if self.__commandline_matcher is None and self.__target_pid is None:
-            raise BadMonitorConfiguration('At least one of the following fields must be provide: commandline or pid',
-                                          'commandline')
+        self.__aggregate_multiple_processes = self._config.get(
+            'aggregate_multiple_processes', default=False, convert_to=bool
+        )
+        self.__include_child_processes = self._config.get(
+            'include_child_processes', default=False, convert_to=bool
+        )
+        if not getattr(self, 'last_discovered', False):
+            # poll interval (in seconds) for matching processes when we are monitoring multiple processes
+            # alive for all epochs of the monitor.
+            self.__last_discovered = {}
+        self.__process_discovery_interval = self._config.get('process_discovery_interval', default=120, convert_to=int)
+        self.__datetime_format = "%Y-%m-%d %H:%M:%S"
```

Comment:
Remove this.  Just keep track of time in milliseconds since epoch.  That's how we do it throughout the rest of the code base for this type of problem.

Reply:
makes sense!